integration/mediation-tests/tests-platform/tests-rabbitmq/src/test/java/org/wso2/carbon/esb/rabbitmq/transport/test/RabbitMQConsumerTestCase.java

```diff
@@ -38,77 +41,67 @@
         logViewer = new LogViewerClient(contextUrls.getBackEndUrl(), getSessionCookie());
     }
 
-    private void initRabbitMQBroker() {
-        sender = new RabbitMQProducerClient("localhost", 5672, "guest", "guest");
+    private void initRabbitMQBroker() throws URISyntaxException {
+        sender = new RabbitMQProducerClient(RabbitMQServerInstance.getHost(), RabbitMQServerInstance.getPort(),
+                "guest", "guest");
         try {
-            sender.declareAndConnect("exchange2", "queue2");
+            sender.declareAndConnect("exchange2", "simple_consumer_test");
         } catch (IOException e) {
             Assert.fail("Could not connect to RabbitMQ broker");
         }
     }
 
-    @Test(groups = {"wso2.esb"}, description = "Test ESB as a RabbitMQ Consumer ")
+    @Test(groups = { "wso2.esb" }, description = "Test ESB as a RabbitMQ Consumer ")
     public void testRabbitMQConsumer() throws Exception {
-        int beforeLogSize = logViewer.getAllRemoteSystemLogs().length;
+        logViewer.clearLogs();
 
-        try {
-            String message =
-                    "<ser:placeOrder xmlns:ser=\"http://services.samples\">\n" +
-                            "<ser:order>\n" +
-                            "<ser:price>100</ser:price>\n" +
-                            "<ser:quantity>2000</ser:quantity>\n" +
-                            "<ser:symbol>RMQ</ser:symbol>\n" +
-                            "</ser:order>\n" +
-                            "</ser:placeOrder>";
-            for (int i = 0; i < 200; i++) {
-                sender.sendMessage(message, "text/plain");
-            }
-        } catch (IOException e) {
-            Assert.fail("Could not connect to RabbitMQ broker");
+        String message = "<ser:placeOrder xmlns:ser=\"http://services.samples\">\n" + "<ser:order>\n"
+                + "<ser:price>100</ser:price>\n" + "<ser:quantity>2000</ser:quantity>\n"
+                + "<ser:symbol>RMQ</ser:symbol>\n" + "</ser:order>\n" + "</ser:placeOrder>";
+        for (int i = 0; i < 200; i++) {
+            sender.sendMessage(message, "text/plain");
         }
 
-        Thread.sleep(20000);
+        RabbitMQTestUtils.waitForLogToGetUpdated();
 
         LogEvent[] logs = logViewer.getAllRemoteSystemLogs();
-        int afterLogSize = logs.length;
         int count = 0;
 
-        for (int i = (afterLogSize - beforeLogSize - 1); i >= 0; i--) {
-            String message = logs[i].getMessage();
-            if (message.contains("received = true")) {
+        for (LogEvent logEvent : logs) {
+            if (logEvent == null) {
+                continue;
+            }
+            String logMessage = logEvent.getMessage();
+            if (logMessage.contains("received = true")) {
                 count++;
             }
         }
-
         Assert.assertEquals(count, 200, "All messages are not received from queue");
```

Comment:
Let us log the count received. Or is it logged when assert fails by test platform?

Reply:
It's by default logged when the assertion fails.